Components/Molecules/GoalItem.js

```diff
@@ -93,11 +94,12 @@ const GoalItem = ({
       },
     })
       .then(() => refetch())
-      .catch(e => console.error(e));
-
-    if (totalCount + 1 == cadenceCount) {
-      goalsListConfetti();
-    }
+      .catch(e => console.error(e))
```

Comment:
catch should be very last so it catches all errors that might happen at any of the previous chains, please move it so it runs after all thens are done.

Reply:
Good feedback - I've addressed and updated